File: Ruby/WeaponType.rb
#encoding:utf-8

# +-------------------------------------------------------------------+ #
# | DEEPSPACE (Ruby)                                                  | #
# | Authors: Shao Jie Hu Chen and Mario Megías Mateo                  | #
# +-------------------------------------------------------------------+ #
# | Enumeration of different weapons and their different programs.    | #
# +-------------------------------------------------------------------+ #

module Deepspace

    module WeaponType

        class Type

            #  Constructor or Initializer
            def initialize (the_power, the_name)
                @power = the_power
                @name  = the_name
            end

            # Get function
            def power
                @power
            end

            # to_s method
            def to_s
                "#{@name}"
            end

        end

        LASER   = Type.new(2.0, "LASER")
        MISSILE = Type.new(3.0, "MISSILE")
        PLASMA  = Type.new(4.0, "PLASMA")
    end

end # end of Deepspace

File: Java/src/WeaponType.java
/**
 * @brief DEEPSPACE (Java). 
 * Enumeration of different weapons and their damage powers. 
 * @author Mario Megías Mateo and Shao Jie Hu Shen.
 */
public enum WeaponType {
    
    LASER(2.0f), MISSILE(3.0f), PLASMA(4.0f);
    
    private final float power;
    
    /**
     * @brief constructor
     * @param p 
     */
    WeaponType(float p) {
        this.power = p;
    }
    
    /**
     * Get method
     * @return 
     */
    float getPower() {
        return this.power;
    } 
}
